maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultPluginValidationManager.java

```diff
@@ -83,13 +83,23 @@ private enum ValidationReportLevel {
     public void onEvent(Object event) {
         if (event instanceof ExecutionEvent) {
             ExecutionEvent executionEvent = (ExecutionEvent) event;
-            if (executionEvent.getType() == ExecutionEvent.Type.SessionEnded) {
+            if (executionEvent.getType() == ExecutionEvent.Type.SessionStarted) {
+                RepositorySystemSession repositorySystemSession =
+                        executionEvent.getSession().getRepositorySession();
+                ValidationReportLevel level = parseValidationReportLevel(repositorySystemSession);
+                repositorySystemSession.getData().set(MAVEN_PLUGIN_VALIDATION_KEY, level);
+            } else if (executionEvent.getType() == ExecutionEvent.Type.SessionEnded) {
                 reportSessionCollectedValidationIssues(executionEvent.getSession());
             }
         }
     }
 
     private ValidationReportLevel validationReportLevel(RepositorySystemSession session) {
+        return (ValidationReportLevel) session.getData()
+                .computeIfAbsent(MAVEN_PLUGIN_VALIDATION_KEY, () -> parseValidationReportLevel(session));
```

Comment:
Can be empty here?
We use the same key and assume that will contains correct type.
If code from onEvent will not execute we can have wrong type.

Reply:
no, this is session.data while string is in session.configProperties, two distinct maps

Reply:
changed to not reuse keys as this may cause misunderstanding